api/src/application/user/user-controller.js

```diff
@@ -110,7 +110,7 @@ module.exports = (
       return (request, reply, source, error) => {
         reply.view('reset-password', {
           title: title,
-          returnTo: `${request.query.redirect_uri}?status=cancelled`,
+          returnTo: (title === 'Set Password') ? false : `${request.query.redirect_uri}?status=cancelled`,
```

Comment:
Let's check `request.query.redirect_uri` rather than `title`, so apps still have the option of having their "Set Password" page redirect if they want and it's a bit more resilient against future changes to "title".